Assets/Scripts/CoinsAmountRenderer.cs

```diff
@@ -0,0 +1,11 @@
+﻿using UnityEngine;
+using TMPro;
+
+public class CoinsAmountRenderer : MonoBehaviour
+{
+    [SerializeField] private Player _player;
+    [SerializeField] private TextMeshProUGUI _coinsText;
+    [SerializeField] private string _template = "Coins: {0}";
+
+    private void Start() => _player.CoinsAmountChanged += () => _coinsText.text = string.Format(_template, _player.Coins);
```

Comment:
https://trello.com/c/QAQLkgbR/7-%D0%B3%D0%B4%D0%B5-%D0%BB%D1%83%D1%87%D1%88%D0%B5-%D0%BF%D0%BE%D0%B4%D0%BF%D0%B8%D1%81%D1%8B%D0%B2%D0%B0%D1%82%D1%8C%D1%81%D1%8F